app/models/discovery_rule.rb

```diff
@@ -40,6 +41,10 @@ def default_int_attributes
    self.priority  ||= 0
   end
 
+  def self.suggest_priority
+    self.unscoped.maximum(:priority).to_i + STEP
```

Comment:
Sorry to jump in so late, but I am wondering won't this make the priority increasing across all organizations? Not a big deal, I wonder if this was meant to be.

Reply:
Yeah, this makes the priority increasing across all orgs. I dont think this is meant to be.
Will fix it in my upcoming pr :)

Reply:
It was meant to behave this way. When this was written, the prority uniqueness validation ignored taxonomy context. If it has changed, the `unscoped` can be removed. But I think the validation still exist or not?

Reply:
Oh yes you are right, we are not there yet!

Reply:
let me try to address rule priority in my pr(https://github.com/theforeman/foreman_discovery/pull/450) itself, works? @ares @lzap

Reply:
Sure

Reply:
Sure, since we are actually relaxing the constraint there is no need of migration now. Just make sure it's suggested correctly.